src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -182,6 +182,12 @@ internal async Task IndexMethodAsyncCore(MethodInfo method, IFunctionIndexCollec
                     {
                         throw new InvalidOperationException("More than one trigger per function is not allowed.");
                     }
+
+                    if ((method.GetCustomAttribute<RetryAttribute>() != null || method.DeclaringType.GetCustomAttribute<RetryAttribute>() != null) &&
```

Comment:
We're already pulling the RetryAttribute in the FromMethod call that CreateFunctiondescriptor makes. So rather than do it twice, recommend determining supportsRetry here and passing that down for the check later.